apps/test.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# Skip tests if the current branch's diff-tree compared to its staging merge-base doesn't touch any files in /apps.
+# Always run tests if the current branch is 'staging'.
+
+BRANCH=$(git rev-parse --abbrev-ref HEAD)
+BRANCH_COMMIT=$(git merge-base HEAD refs/remotes/origin/staging)
+if [ ${BRANCH} == 'staging' ] || git diff-tree HEAD ${BRANCH_COMMIT} --name-only | grep -q "apps$"; then
```

Comment:
My grep-fu is weak - this looks like you are looking for every file that ends in apps? Shouldn't this be begins with?